### DISCUSSION TITLE
Q: `Run a shell command` default arguments

### DISCUSSION THREAD

**Discussion by VKondakoff** [sxyazi/yazi#1217] [1. Q&A] [answered]
Hi!

I'm just experimenting with `Run a shell command`, invoked by pressing `;` or `:` hotkey. Here is what I noticed: everything is OK if I'm writing something like this in the input field (open hovered file with default app): 
`open $0`
But is I omit the argument and input just `open` command stops to work and here is what I see in Task manager:

<img width="1139" alt="Снимок экрана 2024-06-26 в 14 35 25" src="https://github.com/sxyazi/yazi/assets/230824/9254a62a-91e1-44ba-8479-75ca5788dcfa">

So, there are _two_ instances of currently hovered file in command line. Why this happens? Bug? Or am I doing something wrong?

**Comment by sxyazi** [comment, marked as answer]
These two paths are identified as $0 and $1, which means you can use $0 and $1 in the shell. Of course, you don't have to use them, but Yazi always passes them to ensure they are available when needed.

**Reply by VKondakoff** [reply]
Got it, thank you. But I think it would be more user friendly to set the default arguments for `Run a shell command` like this:
1. If there is nothing selected add `$0` to the user specified command
2. If there are some files/folders selected add `$@` to the user specified command

**Reply by sxyazi** [reply]
When no file is selected (only hovering over a file), `$0 == $1`. This provides convenience for the user: for openers that support opening multiple files, `$@` will always be available. Conversely, if only `$0` is available, `$@` will no longer be available since `$0` is not considered part of `$@`, requiring users to write two sets of commands for programs that support both multiple and single paths. On the other hand, `$0` is an extension by Yazi and doesn't belong to any standard, while `$1` is always available, aligning with shell behavior.
